peekingduck/pipeline/nodes/model/osnetv1/osnet_files/osnet.py

```diff
@@ -0,0 +1,307 @@
+    """Omni-scale feature learning block."""
+
+    def __init__(  # pylint: disable=unused-argument
+        self,
+        in_channels: int,
+        out_channels: int,
+        IN: bool = False,
+        bottleneck_reduction: int = 4,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Args:
+            in_channels (int): Number of channels in the input image.
+            out_channels (int): Number of channels produced by the convolution.
+            IN (bool): Applies Instance Normalization. Defaults to False.
+            bottleneck_reduction (int): Reduction rate in the bottleneck
+                architecture. Defaults to 4.
+        """
+        super().__init__()
+        mid_channels = out_channels // bottleneck_reduction
+        self.conv1 = Conv1x1(in_channels, mid_channels)
+        self.conv2a = LightConv3x3(mid_channels, mid_channels)
+        self.conv2b = nn.Sequential(
+            LightConv3x3(mid_channels, mid_channels),
+            LightConv3x3(mid_channels, mid_channels),
+        )
+        self.conv2c = nn.Sequential(
+            LightConv3x3(mid_channels, mid_channels),
+            LightConv3x3(mid_channels, mid_channels),
+            LightConv3x3(mid_channels, mid_channels),
+        )
+        self.conv2d = nn.Sequential(
+            LightConv3x3(mid_channels, mid_channels),
+            LightConv3x3(mid_channels, mid_channels),
+            LightConv3x3(mid_channels, mid_channels),
+            LightConv3x3(mid_channels, mid_channels),
+        )
+        self.gate = ChannelGate(mid_channels)
+        self.conv3 = Conv1x1Linear(mid_channels, out_channels)
+        self.downsample = None
+        if in_channels != out_channels:
+            self.downsample = Conv1x1Linear(in_channels, out_channels)
+        self.IN = None
+        if IN:
+            self.IN = nn.InstanceNorm2d(out_channels, affine=True)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Defines the computation performed at every call."""
+        identity = x
+        x1 = self.conv1(x)
+        x2a = self.conv2a(x1)
+        x2b = self.conv2b(x1)
+        x2c = self.conv2c(x1)
+        x2d = self.conv2d(x1)
+        x2 = self.gate(x2a) + self.gate(x2b) + self.gate(x2c) + self.gate(x2d)
+        x3 = self.conv3(x2)
+        if self.downsample is not None:
+            identity = self.downsample(identity)
+        out = x3 + identity
+        if self.IN is not None:
+            out = self.IN(out)
+        return F.relu(out)
+
+
+# Network architecture
+class OSNet(nn.Module):  # pylint: disable=too-many-instance-attributes
+    """Omni-Scale Network."""
+
+    def __init__(  # pylint: disable=too-many-arguments, unused-argument
+        self,
+        num_classes: int,
+        blocks: List[OSBlock],
+        layers: List[int],
+        channels: List[int],
+        feature_dim: int = 512,
+        loss: str = "softmax",
+        IN: bool = False,
```

Comment:
Similar to above this shouldn't be all caps

Reply:
Renamed